internal/controller/object/object.go

```diff
@@ -223,9 +223,6 @@ func (c *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 		return managed.ExternalCreation{}, err
 	}
 
-	obj.SetAnnotations(map[string]string{
-		v1.LastAppliedConfigAnnotation: string(cr.Spec.ForProvider.Manifest.Raw),
-	})
```

Comment:
If we delete these lines, then we won't set the last applied configuration annotation that provider-kubernetes uses to decide whether an update is needed or not. 
We should instead check existing annotations and decide to append or build a new one with last applied config.

Reply:
on my Side the last applied configruation  annotation is by default - let me check

Reply:
yeah, noticed your PR description but there no code piece left to set that. Still tried on my side and verified no last applied config set with this PR.

Reply:
@turkenh  added a version to merge the annotations - can you have a look ? thanks

Reply:
@haarchri just found that crossplane-runtime already has a method for that which we can just reuse: https://github.com/crossplane/crossplane-runtime/blob/master/pkg/meta/meta.go#L201

Reply:
@turkenh  thanks - i changed it to addAnnotation from crossplane-runtime - thanks

Reply:
great, thanks!